framer/Layer.coffee

```diff
@@ -882,6 +883,14 @@ class exports.Layer extends BaseClass
 			@_getPropertyValue "image"
 		set: (value) ->
 
+			if LinearGradient.isLinearGradient(value)
+				oldValue = @_getPropertyValue "image"
```

Comment:
You can move the currentValue on line 897 below above the if statement and use that.